src/trace/segment.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Base64 } from 'js-base64';
+import xhrInterceptor from '../interceptors/xhr';
+import uuid from '../services/uuid';
+import Report from '../services/report';
+import { SegmentFeilds, SpanFeilds } from './type';
+import { SpanLayer, SpanType } from '../services/constant';
+import { CustomOptionsType } from '../types';
+
+export default function traceSegment(options: CustomOptionsType) {
+  const segment = {
+    traceId: uuid(),
+    service: options.service,
+    spans: [],
+    serviceInstance: options.serviceVersion,
+    traceSegmentId: options.segmentId,
+  } as SegmentFeilds;
+  const segCollector: { event: XMLHttpRequest; startTime: number }[] | any = [];
+  // inject interceptor
+  xhrInterceptor();
+  window.addEventListener('xhrReadyStateChange', (event: CustomEvent) => {
+    const xhrState = event.detail.readyState;
+
+    if (xhrState === 1) {
```

Comment:
Could you add comments about the meaning of `xhrState` values? And add constant to represent them?

Reply:
Sure, let me work on it a bit